projects/igniteui-angular/src/lib/expansion-panel/toggle-animation-component.ts

```diff
@@ -0,0 +1,190 @@
+import { AnimationBuilder, AnimationPlayer, AnimationReferenceMetadata, useAnimation } from '@angular/animations';
+import { Directive, ElementRef, EventEmitter, OnDestroy } from '@angular/core';
+import { noop, Subject } from 'rxjs';
+import { growVerIn, growVerOut } from '../animations/grow';
+
+/**@hidden @internal */
+export interface ToggleAnimationSettings {
+    openAnimation: AnimationReferenceMetadata;
+    closeAnimation: AnimationReferenceMetadata;
+}
+
+export interface ToggleAnimationOwner {
```

Comment:
Thanks! I completely agree. If we'd like to reuse this logic in the overlay as well, we'd need to abstract more of the logic. There would still be the need for a `ToggleAnimationOwner` for the handling of reversing / clearing opposite animations, but the init logic should be in a separate class.
IMO it should be done in a separate PR / issue. The only reason part of it is in this PR is so that we can reuse the animation logic of the expansion panel.